crates/solvers/src/boundary/legacy.rs

```diff
@@ -578,6 +581,22 @@ fn to_boundary_auction_result(notification: &notification::Notification) -> (i64
 
     let auction_result = match &notification.kind {
         Kind::EmptySolution => AuctionResult::Rejected(SolverRejectionReason::NoUserOrders),
+        Kind::SimulationFailed(tx) => AuctionResult::Rejected(
+            SolverRejectionReason::SimulationFailure(TransactionWithError {
+                error: "".to_string(),
+                transaction: SimulatedTransaction {
+                    from: tx.from.into(),
+                    to: tx.to.into(),
+                    data: tx.input.clone().into(),
+                    internalization: InternalizationStrategy::SkipInternalizableInteraction,
```

Comment:
I think the internalization flag should probably not be hardcoded. The simulation could also only fail if internalized AFAIK.

Reply:
I aimed here to actually drop the `internalization` flag as it is not needed IMO, since all solver engines are sending raw interactions to the `driver` and, in case of simulation failure, they get the failed transaction with calldata, so that would be enough to see if it was internalized or not (if the calldata in notification differs from what they sent to us, and they have `solution id` to match on).

So, I just want to be careful and add the bare minimum of data needed (to reduce complexity).

Here, I would rather add additional variant `InternalizationStrategy::Undefined` or similar, to make the code accurate and not misleading, instead of providing actual value.

WDYT?

Reply:
Added `InternalizationStrategy::Unknown`

Reply:
Sorry, saw the follow up comments too late.
Wouldn't it then make more sense to drop the `internalization` here altogether? Seems like in all the existing use cases having the `Unknown` variant doesn't really make sense and therefore compromising the use of an existing type for this doesn't seem worth it.

Reply:
Yes we can definitely drop it. Maybe I was too cautious to not cause any backward compatibility issues since this is after all a debugging feature.

Reply:
Tried removing but we are using this field to recreate tenderly link in the non-colocated driver also, so we will have to remove it only once the colocation is activated. 
I added the comment in the code to remember this.